samples/chat-app/server/Schema.fs

```diff
@@ -0,0 +1,759 @@
+      Name = x.Name }
+
+  let chatMemberInDb_To_ChatMember (membersToGetDetailsFrom : Member_In_Db seq) (x : ChatMember_In_Db) : ChatMember =
+    let memberDetails = membersToGetDetailsFrom |> Seq.find (fun m -> m.Id = x.MemberId)
+    { Id = x.MemberId
+      Name = memberDetails.Name
+      Role = x.Role }
+
+  let chatMemberInDb_To_MeAsAChatMember (membersToGetDetailsFrom : Member_In_Db seq) (x : ChatMember_In_Db) : MeAsAChatMember =
+    let memberDetails = membersToGetDetailsFrom |> Seq.find (fun m -> m.Id = x.MemberId)
+    { PrivId = memberDetails.PrivId
+      Id = x.MemberId
+      Name = memberDetails.Name 
+      Role = x.Role }
+
+  let chatRoomInDb_To_ChatRoom (membersToGetDetailsFrom : Member_In_Db seq) (x : ChatRoom_In_Db) : ChatRoom =
+    { Id = x.Id
+      Name = x.Name
+      Members =
+        FakePersistence.ChatMembers.Values
+        |> Seq.filter (fun m -> x.Members |> List.contains m.MemberId)
+        |> Seq.map (chatMemberInDb_To_ChatMember membersToGetDetailsFrom)
+        |> List.ofSeq }
+
+  let chatRoomInDb_To_ChatRoomForMember (membersToGetDetailsFrom : Member_In_Db seq) (chatMember : ChatMember_In_Db) (x : ChatRoom_In_Db) : ChatRoomForMember =
+    { Id = x.Id
+      Name = x.Name
+      MeAsAChatMember = chatMember |> chatMemberInDb_To_MeAsAChatMember membersToGetDetailsFrom
+      OtherChatMembers =
+        FakePersistence.ChatMembers.Values
+        |> Seq.filter (fun m -> m.MemberId <> chatMember.MemberId && x.Members |> List.contains m.MemberId)
+        |> Seq.map (chatMemberInDb_To_ChatMember membersToGetDetailsFrom)
+        |> List.ofSeq }
+
+  let organizationInDb_To_Organization (x : Organization_In_Db) : Organization =
+    let members =
+      FakePersistence.Members.Values
+        |> Seq.filter (fun m -> x.Members |> List.contains m.Id)
+    { Id = x.Id
+      Name = x.Name
+      Members = members |> Seq.map memberInDb_To_Member |> List.ofSeq
+      ChatRooms =
+        FakePersistence.ChatRooms.Values
+        |> Seq.filter (fun c -> x.ChatRooms |> List.contains c.Id)
+        |> Seq.map (chatRoomInDb_To_ChatRoom members)
+        |> List.ofSeq }
+
+  let organizationInDb_To_OrganizationForMember (memberId : MemberId) (x : Organization_In_Db) : OrganizationForMember option =
+    let mapToOrganizationForMemberForMember (memberInDb : Member_In_Db) =
+      let organizationStats = x |> organizationInDb_To_Organization
+      { OrganizationForMember.Id =  x.Id
+        Name = x.Name
+        MeAsAMember = memberInDb |> memberInDb_To_MeAsAMember
+        OtherMembers = organizationStats.Members |> List.filter(fun m -> m.Id <> memberInDb.Id)
+        ChatRooms = organizationStats.ChatRooms }
+    FakePersistence.Members.Values
+    |> Seq.tryFind (fun m -> m.Id = memberId)
+    |> Option.map mapToOrganizationForMemberForMember
+
+
+module Schema =
+  open FSharp.Data.GraphQL.Server.AppInfrastructure.Rop
+
+  let validationException_Member_With_This_Name_Already_Exists (theName : string) =
```

Comment:
Could we put this into an `Exceptions` sub-module and remove the `validationException` prefix, maybe?

Reply:
Done.